internal/jsre/deps/web3.js

```diff
@@ -3706,6 +3706,13 @@ var inputDefaultBlockNumberFormatter = function (blockNumber) {
     return inputBlockNumberFormatter(blockNumber);
 };
 
+var inputBlockNumberOrHashFormatter = function (numberOrHash) {
```

Comment:
Yeah not sure about changing the dependency library directly.

Reply:
Which one will succeed, ethers.js, web3.js or another package?

Reply:
We should just not directly change a 3rd party library in our code.

As for your question: we're now at the latest version of web3 that doesn't use Promises and is sync. We'd like to maintain the ease of working with sync methods and chaining them etc.
It'd be nice to upgrade to ethers.js at some point if we can make it seamless for users. Our JS vm (goja) has support for promises and async/await. However ideally we should avoid for users to do promise handling in console all the time.